db/migrate/20190729194551_change_network_adapters.rb

```diff
@@ -0,0 +1,54 @@
+class ChangeNetworkAdapters < ActiveRecord::Migration[5.2]
+  def change
+    add_reference :network_adapters, :source, :index => false, :null => false, :foreign_key => {:on_delete => :cascade}
+    add_reference :network_adapters, :orchestration_stack, :index => true, :null => true, :foreign_key => {:on_delete => :cascade}
+
+    # Relations to VMs, Hosts, etc.
+    add_reference :network_adapters, :device, :index => true, :polymorphic => true, :null => true
```

Comment:
@agrare so I have the other approach commented below, but for now I think I prefer the polymorphic approach. Let me try how that looks with the persister schema. The other approach would need a lot of unique indexes, if we'd add mode devices. Also the relation is just `{:on_delete => :nullify}` so it's not a big deal that it's polymorphic

Reply:
Yeah I agree the polymorphic assoc seems to be the best if we can make it work